packages/flutter/lib/src/painting/_network_image_web.dart

```diff
@@ -0,0 +1,72 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:ui' show hashValues;
+import 'dart:ui' as ui;
+
+import 'package:flutter/foundation.dart';
+
+import 'image_provider.dart' as image_provider;
+import 'image_stream.dart';
+
+/// The dart:html implemenation of [image_provider.NetworkImage].
+class NetworkImage extends image_provider.ImageProvider<image_provider.NetworkImage> implements image_provider.NetworkImage {
+  /// Creates an object that fetches the image at the given URL.
+  ///
+  /// The arguments must not be null.
+  const NetworkImage(this.url, {this.scale = 1.0, this.headers})
```

Comment:
cc @kevmoo - This is an interesting case you brought up today, headers do nothing on the web!